Common/Systems/PotionPlayer.cs

```diff
@@ -51,6 +59,15 @@ internal static void UseHealingPotion(Player self, bool hasSync = false)
 		// Changed from flat healing to percentage-based (Using statLifeMax2 is MaxLife after adjusted buffs/equipment)
 		int healAmount = (int)(self.statLifeMax2 * (mp.HealPower / 100f));
 		self.Heal(healAmount);
+
+		foreach (ModPlayer plr in self.ModPlayers)
+		{
+			if (plr is IOnCustomPotionPlayer customPot)
+			{
+				customPot.OnCustomPotion(true, healAmount);
```

Comment:
Use a HookList as seen here:
https://github.com/Mirsario/TerrariaOverhaul/blob/dev/Common/Movement/_Hooks/IPlayerOnBunnyhopHook.cs
ModPlayers iteration may contain like three hundred instances nowadays.
(Making a separate file is not necessary, wouldn't have done that today.)